### PR TITLE
KOGITO-2089: Task Form Building Block integration on Task Inbox

### PR DESCRIPTION
**Pull request by pefernan**
JIRA: https://issues.redhat.com/browse/KOGITO-2089

- Improved details page following UX designs.
- Improved details header
- Changed the submit notifications from modals to alerts
- Improved interactions, making the user stay in the details page after submitting the form.
- Added empty state in case there's no selected task

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by pefernan** [comment]
@nmirasch @jstastny-cz @AjayJagan your turn!!

File: ui-packages/packages/task-console/server/MockData/controllers.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ module.exports = controller = {
     });
 
     if(tasksUnableToTransition.includes(task.id)) {
-      res.status(500).send("");
+      res.status(500).send("Unexpected failure when doing a transition!");
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/server/MockData/controllers.js]: should enforce `'` over `"` in string literals.

File: ui-packages/packages/task-console/src/components/Atoms/FormNotification/FormNotification.tsx [hunk 2]
```diff
@@ -14,38 +14,72 @@
...
+export interface Action {
+  label: string;
+  onClick: () => void;
+}
+
+interface IOwnProps {
+  notification: Notification;
+}
+
+const FormNotification: React.FC<IOwnProps> = ({ notification }) => {
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Atoms/FormNotification/FormNotification.tsx]: why did you remove the OUIAProps? I mean, in task-console we don't have yet the coverage, so it is not blocking issue.

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx [hunk 3]
```diff
@@ -15,40 +15,68 @@
...
-import { GraphQL } from '@kogito-apps/common';
-import UserTaskInstance = GraphQL.UserTaskInstance;
+enum Variant {
+  INLINE = 'inline',
+  LABEL = 'label'
+}
 
 interface IOwnProps {
   task: UserTaskInstance;
+  variant?: 'inline' | 'label';
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx]: shouldn't we reference here the enum from above? `variant?: Variant`

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx [hunk 4]
```diff
@@ -15,40 +15,68 @@
...
+import { Label } from '@patternfly/react-core';
 import {
   BanIcon,
   CheckCircleIcon,
   OnRunningIcon
 } from '@patternfly/react-icons';
 
-import { GraphQL } from '@kogito-apps/common';
-import UserTaskInstance = GraphQL.UserTaskInstance;
+enum Variant {
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx]: I think we might need to export this too, cause it should be part of props passed to the component from another component.

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/tests/TaskState.test.tsx [hunk 5]
```diff
@@ -102,6 +111,67 @@ describe('TaskState', () => {
...
+    expect(label.exists()).toBeFalsy();
+
+    const icon = wrapper.find(CheckCircleIcon);
+
+    expect(icon.exists()).toBeTruthy();
+  });
+
+  it('Test show active task in label', () => {
+    const wrapper = getWrapper(
+      <TaskState task={userTask} variant={'label'} />,
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Atoms/TaskState/tests/TaskState.test.tsx]: and here we should IMO pass the enum Variant value.

File: ui-packages/packages/task-console/src/components/Molecules/PageTitle/PageTitle.tsx [hunk 6]
```diff
@@ -1,16 +1,34 @@
...
+import { Flex, FlexItem, Title } from '@patternfly/react-core';
+import { componentOuiaProps, OUIAProps } from '@kogito-apps/common';
 
 export interface IOwnProps {
   title: any;
+  extra?: any;
 }
 
-const PageTitle: React.FC<IOwnProps> = ({ title }) => {
+const PageTitle: React.FC<IOwnProps & OUIAProps> = ({
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Molecules/PageTitle/PageTitle.tsx:10]: ok, now it seems that you do want to introduce the OUIA attributes.

File: ui-packages/packages/task-console/src/components/Molecules/PageTitle/PageTitle.tsx [hunk 7]
```diff
@@ -1,16 +1,34 @@
...
     <React.Fragment>
-      <Title headingLevel="h1" size="4xl">
-        {title}
-      </Title>
+      <Flex>
+        <FlexItem spacer={{ default: 'spacerSm' }}>
+          <Title
+            headingLevel="h1"
+            size="4xl"
+            {...componentOuiaProps(ouiaId, 'page-title', ouiaSafe)}
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Molecules/PageTitle/PageTitle.tsx]: the ouia attributes need to be at top-level element of the defining component (when rendered) - so I'd say the Flex here, cause fragment can't have attributes and Flex is the first actually rendering sth.

File: ui-packages/packages/task-console/src/components/Atoms/FormNotification/FormNotification.tsx [hunk 8]
```diff
@@ -14,38 +14,72 @@
...
+  let variant;
+
+  const [showDetails, setShowDetails]: [boolean, any] = useState(false);
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Atoms/FormNotification/FormNotification.tsx]: @pefernan  - you could do something like this for types - 
`const [showDetails, setShowDetails] = useState<boolean>(false);`

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx [hunk 9]
```diff
@@ -15,40 +15,68 @@
...
 
 interface IOwnProps {
   task: UserTaskInstance;
+  variant?: 'inline' | 'label';
 }
 
-const TaskState: React.FC<IOwnProps> = ({ task }) => {
-  let icon;
+const TaskState: React.FC<IOwnProps> = ({ task, variant }) => {
+  const icon = resolveTaskStateIcon(task);
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx]: hey @pefernan, can you try using this- `const icon:JSX.Element = resolveTaskStateIcon(task);`

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx [hunk 10]
```diff
@@ -15,40 +15,68 @@
...
+
+  if (variant === Variant.LABEL) {
+    const color = resolveTaskStateLabelColor(task);
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx:38]: try this ?-`const color:string = resolveTaskStateLabelColor(task);`

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx [hunk 11]
```diff
@@ -15,40 +15,68 @@
...
+  }
+
+  return (
+    <React.Fragment>
+      {icon} {task.state}
+    </React.Fragment>
+  );
+};
 
+function resolveTaskStateIcon(task: UserTaskInstance) {
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx]: try this? - `function resolveTaskStateIcon(task: UserTaskInstance):JSX.Element {`

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx [hunk 12]
```diff
@@ -15,40 +15,68 @@
...
   }
+}
 
-  return (
-    <React.Fragment>
-      {icon} {task.state}
-    </React.Fragment>
-  );
-};
+function resolveTaskStateLabelColor(task: UserTaskInstance) {
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx:68]: try this? - `function resolveTaskStateLabelColor(task: UserTaskInstance):string {`

File: ui-packages/packages/task-console/src/components/Organisms/TaskForm/TaskForm.tsx [hunk 13]
```diff
@@ -1,41 +1,57 @@
...
   const context: IContext<UserTaskInstance> = useContext(TaskConsoleContext);
-  const [alertMessage, setAlertMessage]: [AlertMessage, any] = useState(null);
-  const [loading, setLoading] = useState(true);
+  const [notification, setNotification]: [Notification, any] = useState(null);
+  const [loading, setLoading]: [boolean, any] = useState(true);
+  const [isSubmitting, setIsSubmitting]: [boolean, any] = useState(false);
+  const [submitted, setSubmitted]: [boolean, any] = useState(false);
   const [stateUserTask, setStateUserTask]: [UserTaskInstance, any] = useState(
     null
   );
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Organisms/TaskForm/TaskForm.tsx]: @pefernan - for all the useState usage we could use something like this - 
`const [state,setState]=useState<type>(value)` - easier for us to read :P

File: ui-packages/packages/task-console/src/components/Templates/UserTaskInstanceDetailsPage/UserTaskInstanceDetailsPage.tsx [hunk 14]
```diff
@@ -32,7 +37,12 @@ const UserTaskInstanceDetailsPage: React.FC<RouteComponentProps<
   {}
 > &
   OUIAProps> = ({ ouiaId, ouiaSafe, ...props }) => {
-  const id = props.match.params.taskID;
+  const id = props.match.params.taskId;
+
+  const [userTask, setUserTask]: [
+    UserTaskInstance,
+    (UserTaskInstance) => void
+  ] = useState();
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Templates/UserTaskInstanceDetailsPage/UserTaskInstanceDetailsPage.tsx]: same here :) - `const [userTask, setUserTask] = useState<UserTaskInstance>();`

File: ui-packages/packages/task-console/src/components/Molecules/PageTitle/PageTitle.tsx [hunk 15]
```diff
@@ -1,16 +1,34 @@
 import React from 'react';
-import { Title } from '@patternfly/react-core';
+import { Flex, FlexItem, Title } from '@patternfly/react-core';
+import { componentOuiaProps, OUIAProps } from '@kogito-apps/common';
 
 export interface IOwnProps {
   title: any;
+  extra?: any;
```
Comment (AjayJagan) [inline ui-packages/packages/task-console/src/components/Molecules/PageTitle/PageTitle.tsx]: maybe use this :
```
 title: string;
  extra?: JSX.Element;
```

File: ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx (hunk 10 above)
Comment (pefernan) [reply ui-packages/packages/task-console/src/components/Atoms/TaskState/TaskState.tsx:38]: In this case I cannot do it, but I agree the change is a good one

**Comment by pefernan** [comment]
Thanks @jstastny-cz & @AjayJagan for your notes, just updated the code

**Review by AjayJagan — APPROVED** [review]
LGTM :)

**Review by jstastny-cz — APPROVED** [review]
